### PR TITLE
fix: Calculated expression error

### PR DESCRIPTION
**Pull request by jiuker**
## Description

This computational expression looks like it has some errors.

## Motivation and Context

fix it

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

File: cmd/format-erasure.go [hunk 1]
```diff
@@ -646,7 +646,7 @@ func saveFormatErasureAll(ctx context.Context, storageDisks []StorageAPI, format
 		}, index)
 	}
 
-	writeQuorum := (len(storageDisks) + 1/2)
+	writeQuorum := ((len(storageDisks) + 1) / 2)
```
Comment (harshavardhana) [inline cmd/format-erasure.go]: In-fact the code is correct @jiuker we do not support partial `format.json` writes anymore and this code is executed when its a fresh setup. 

So this code is correct and it should if all drives didn't succeed not 50% of them.  So please change the code to be 

```
reduceWriteQuorumErrs(ctx, g.Wait(), nil, len(storageDisks))
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by jiuker** [comment]
review,plz, @harshavardhana

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#c27c54bb-ae4a-457e-97c2-e86ca1feba92) |
| mint-pools.sh | :x: [more...](#5f2779ac-2877-4583-84d5-3ce156d87883) |
<a name="5f2779ac-2877-4583-84d5-3ce156d87883"></a>
### 15362-64f1d9a/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s3:31639
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 55 seconds
(2/14) Running aws-sdk-go tests ... done in 2 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 44 seconds
(5/14) Running aws-sdk-ruby tests ... done in 6 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... FAILED in 1 minutes and 0 seconds
{
  "name": "mc",
  "duration": "10115",
  "function": "teardown",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-22685 --quiet --no-color rb --force myminio/mc-test-bucket-21938  >>> \nmc: <ERROR> Unable to validate target `myminio/mc-test-bucket-21938`. Get \"http://minio-k8s3:31639/mc-test-bucket-21938/?location=\": dial tcp 15.15.15.3:31639: i/o timeout"
}
(7/14) Running minio-go tests ... done in 1 minutes and 48 seconds
(8/14) Running minio-java tests ... done in 35 seconds
(9/14) Running minio-js tests ... done in 1 minutes and 7 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 37 seconds
(11/14) Running s3cmd tests ... done in 25 seconds
(12/14) Running s3select tests ... done in 5 seconds
(13/14) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 14 tests successfully.
```
<a name="c27c54bb-ae4a-457e-97c2-e86ca1feba92"></a>
### 15362-64f1d9a/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:31468
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 53 seconds
(2/14) Running aws-sdk-go tests ... done in 1 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 44 seconds
(5/14) Running aws-sdk-ruby tests ... done in 6 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 34 seconds
{
  "name": "mc",
  "duration": "24813",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-14399 --quiet --no-color rm mc-test-object-6711.downloaded myminio/mc-test-bucket-16901/mc-test-object-6711  >>> \nRemoving ``.\nmc: <ERROR> Failed to remove `myminio/mc-test-bucket-16901/mc-test-object-6711`. Get \"http://minio-k8s6:31468/mc-test-bucket-16901/?location=\": dial tcp 15.15.15.6:31468: i/o timeout"
}
(7/14) Running minio-go tests ... done in 1 minutes and 49 seconds
(8/14) Running minio-java tests ... done in 34 seconds
(9/14) Running minio-js tests ... done in 1 minutes and 6 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 22 seconds
(11/14) Running s3cmd tests ... done in 24 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :x: [more...](#92ae6f41-a66d-4ca8-bdcb-dbe7b105909a) |
<a name="92ae6f41-a66d-4ca8-bdcb-dbe7b105909a"></a>
### 15362-d6e2285/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:31347
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 57 seconds
(2/14) Running aws-sdk-go tests ... done in 1 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 43 seconds
(5/14) Running aws-sdk-ruby tests ... done in 6 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 40 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 23 seconds
(9/14) Running minio-java tests ... done in 37 seconds
(10/14) Running minio-js tests ... FAILED in 1 minutes and 45 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 2,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"
}
(10/14) Running minio-py tests ... done in 1 minutes and 18 seconds
(11/14) Running s3cmd tests ... done in 26 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#48840a48-e730-4a67-9dc5-64ebd56d2e7d) |
| mint-dist-erasure.sh | :x: [more...](#c8b198e7-7411-4981-a5fc-d3b324e3dc49) |
<a name="c8b198e7-7411-4981-a5fc-d3b324e3dc49"></a>
### 15362-b0c4200/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:30459
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 56 seconds
(2/14) Running aws-sdk-go tests ... done in 1 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 44 seconds
(5/14) Running aws-sdk-ruby tests ... done in 6 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 27 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 12 seconds
(9/14) Running minio-java tests ... FAILED in 5 minutes and 30 seconds
{
  "name": "minio-java",
  "function": "getPresignedPostFormData()",
  "duration": 300028,
  "status": "FAIL",
  "error": "java.net.SocketTimeoutException: timeout >>> [okio.SocketAsyncTimeout.newTimeoutException(JvmOkio.kt:143), okio.AsyncTimeout.access$newTimeoutException(AsyncTimeout.kt:162), okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:340), okio.RealBufferedSource.indexOf(RealBufferedSource.kt:449), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:333), okhttp3.internal.http1.HeadersReader.readLine(HeadersReader.kt:29), okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:178), okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:106), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:79), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), FunctionalTest.getPresignedPostFormData(FunctionalTest.java:1756), FunctionalTest.runObjectTests(FunctionalTest.java:3762), FunctionalTest.runTests(FunctionalTest.java:3781), FunctionalTest.runEndpointTests(FunctionalTest.java:3881), FunctionalTest.main(FunctionalTest.java:4002)]"
}
(9/14) Running minio-js tests ... done in 1 minutes and 3 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 18 seconds
(11/14) Running s3cmd tests ... done in 25 seconds
(12/14) Running s3select tests ... done in 3 seconds
(13/14) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 13 out of 14 tests successfully.
```
<a name="48840a48-e730-4a67-9dc5-64ebd56d2e7d"></a>
### 15362-b0c4200/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:32180
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 54 seconds
(2/14) Running aws-sdk-go tests ... done in 1 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 44 seconds
(5/14) Running aws-sdk-ruby tests ... done in 7 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 26 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 14 seconds
(9/14) Running minio-java tests ... done in 32 seconds
(10/14) Running minio-js tests ... FAILED in 1 minutes and 47 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"Bucket Versioning API\"",
  "duration": 8,
  "status": "FAIL",
  "error": "S3Error: The specified bucket does not exist at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(10/14) Running minio-py tests ... done in 1 minutes and 22 seconds
(11/14) Running s3cmd tests ... done in 24 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Pull request merged**